lib/liquid/standardfilters.rb

```diff
@@ -284,11 +284,13 @@ def truncatewords(input, words = 15, truncate_string = "...")
     # @liquid_type filter
     # @liquid_category string
     # @liquid_summary
-    #   Splits a string into an array of substrings based on a given separator.
+    #   Splits a string into an array of substrings based on a given separator and an optional limit number
     # @liquid_syntax string | split: string
     # @liquid_return [array[string]]
-    def split(input, pattern)
-      input.to_s.split(pattern.to_s)
+    def split(input, pattern, limit = nil)
+      limit = limit.respond_to?(:to_i) ? limit.to_i : 0
```

Comment:
I think we should use `Liquid::Utils.to_integer` here.

Reply:
Oh! I didn't know. I will update the PR ASAP.

Reply:
I made the requested change. Also I updated the method docs.

Reply:
After thinking about it, I think you're right. `Utils.to_integer` is better here. Let's add a test that it raises a `Liquid::ArgumentError` when a non-int is passed as well.

Reply:
Great. I made the required changes.